presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -1372,6 +1316,88 @@ public AnalyzePropertyManager getAnalyzePropertyManager()
         return analyzePropertyManager;
     }
 
+    @Override
+    public MetadataResolver getMetadataResolver(Session session)
```

Comment:
Is this is all here anyway, what was gained by extracting the interface? Is there new functionality somewhere I'm missing?

Reply:
Good question. We are looking at the lighter version of the metadata interface which could be used by the analyzers later on. Right now the metadata interface is available in the presto-main and it has lots of dependencies. By extracting a lighter version (MetadataResolver) and making it available in the presto-analyzer allows analyzers to access minimal required metadata without too many dependencies.